app/routing/routing_engine.py

```diff
@@ -19,22 +26,62 @@ def get_next_location(self, current_location):
             if current_block:
                 current_group = current_block.container
 
-                for index, block in enumerate(current_group.blocks):
-                    if block.id == current_block.id:
-                        if index + 1 < len(current_group.blocks):
-                            # return the next block in this group
-                            return current_group.blocks[index + 1].id
-                for index, group in enumerate(self._schema.groups):
-                    if group.id == current_group.id:
-                        if index + 1 < len(self._schema.groups):
-                            # return the first block in the next group
-                            return self._schema.groups[index + 1].blocks[0].id
-
-                # There are no more blocks or groups, go to summary
-                next_location = 'summary'
+                # Check if there are any routing rules
+                routing_rules = current_block.routing_rules
+                if routing_rules:
+                    return self.process_routing_rules(routing_rules)
+                else:
+                    next_location = self.basic_routing(current_group, current_block)
             else:
                 raise RoutingException('Cannot route: No current block')
         return next_location
 
     def get_first_block(self):
         return self._schema.groups[0].blocks[0].id
+
+    def basic_routing(self, current_group, current_block):
+        # function for linear routing, i.e it goes to the next block in the JSON
+        for index, block in enumerate(current_group.blocks):
+            if block.id == current_block.id:
+                    if index + 1 < len(current_group.blocks):
+                        # return the next block in this group
+                        return current_group.blocks[index + 1].id
+        for index, group in enumerate(self._schema.groups):
+            if group.id == current_group.id:
+                    if index + 1 < len(self._schema.groups):
+                        # return the first block in the next group
+                        return self._schema.groups[index + 1].blocks[0].id
+        # There are no more blocks or groups, go to summary
+        return 'summary'
+
+    @staticmethod
+    def process_routing_rules(routing_rules):
+
+        logger.debug("Processing routing rules", routing_rules)
```

Comment:
No - Although perhaps we should look again at our Flake8 usage, this is a good example of how to configure it: https://bugzilla.mozilla.org/show_bug.cgi?id=1237977